beacon-chain/rpc/slasher/client.go

```diff
@@ -0,0 +1,99 @@
+package slasher
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/beacon-chain/blockchain"
+	"github.com/prysmaticlabs/prysm/beacon-chain/operations/slashings"
+	"github.com/prysmaticlabs/prysm/beacon-chain/p2p"
+	st "github.com/prysmaticlabs/prysm/beacon-chain/state"
+	slashpb "github.com/prysmaticlabs/prysm/proto/slashing"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var log logrus.FieldLogger
+
+func init() {
+	log = logrus.WithField("prefix", "rpc/slasher")
+}
+
+// Client defines a client implementation of the gRPC slasher service.
+type Client struct {
+	HeadFetcher     blockchain.HeadFetcher
+	SlashingPool    *slashings.Pool
+	SlasherClient   slashpb.SlasherClient
+	P2p             p2p.Broadcaster //P2p Will later be used to send slashing on pub sub.
+	ShouldBroadcast bool
+}
+
+// SlashingPoolFeeder this is a stub for the coming PRs #3133
+// Store validator index to public key map Validate attestation signature.
+func (s *Client) SlashingPoolFeeder(ctx context.Context) error {
+	secondsPerEpoch := params.BeaconConfig().SecondsPerSlot * params.BeaconConfig().SlotsPerEpoch
+	d := time.Duration(secondsPerEpoch) * time.Second
+	tick := time.Tick(d)
+	log.Infof("started slasher ticker %v", d)
+	for {
+		select {
+		case <-tick:
+			log.Info("updating slashing pool...")
+			if err := s.updatePool(ctx); err != nil {
+				return errors.Wrap(err, "failed to update slashing pool")
+			}
+		case <-ctx.Done():
+			err := status.Error(codes.Canceled, "Stream context canceled")
+			log.WithError(err)
+			return err
+		}
+	}
+}
+
+func (s *Client) updatePool(ctx context.Context) error {
+	state, err := s.HeadFetcher.HeadState(ctx)
+	if err != nil {
+		return status.Errorf(codes.Internal, "Could not get head state: %v", err)
+	}
+
+	if s.SlasherClient != nil {
+		if err := s.updateSlashingPool(ctx, state); err != nil {
+			log.WithError(err)
```

Comment:
Refrain from using `log.WithError()`, it doesn't log or print out the error at all, use `log.Error()` instead.
Also, you should only return the error when you should end the current process, and log it out if the process should continue regardless of the error.